AlpsSDK/ProximityHandler.swift

```diff
@@ -10,224 +10,127 @@ import Foundation
-                }
+    private func handleProximity(key: String, value: [IBeaconTriple]) {
+        let distance = setUp(key: key)
+        value.forEach { (iBeaconTriple) in
+            guard let deviceId = iBeaconTriple.deviceId else {return}
+            guard let keys = beaconsTriggered[key]?.keys else {return}
+            if keys.contains(deviceId) {
+                // do refresh
+                refreshTriggers(key: key, deviceId: deviceId, distance: distance)
+            } else {
+                // do trigger
+                triggers(key: key, deviceId: deviceId, distance: distance)
             }
         }
     }
     
-    // Re-launch a proximity event depending on refreshTimer (Default : 60 seconds)
-    func refreshTriggers() {
-        for clProximity in CLProximity.allValues {
-            var (beacons, trigger, distance) = setUpTriggerBeaconsProximityEvent(forCLProximity: clProximity)
-            for id in beacons {
-                // Check if the existed proximity event needs a refresh on a based timer
-                let proximityEvent = trigger[id]
-                // Represents  the UNIX current time in milliseconds
-                let now = Int64(Date().timeIntervalSince1970 * 1000)
-                guard let proximityEventCreatedAt = proximityEvent?.createdAt else {
-                    return
-                }
-                let gap = now - proximityEventCreatedAt
-                let truncatedGap = Int(truncatingBitPattern: gap)
-                if truncatedGap > refreshTimer {
-                    // Send the refreshing proximity event based on the timer
-                    let newProximityEvent = ProximityEvent.init(deviceId: id, distance: distance)
-                    let deviceId = "NEED CHANGE"
-                    sendProximityEvent(deviceId: deviceId, proximityEvent: newProximityEvent) {(_ proximityEvent) in
-                        guard let pe = proximityEvent else {
-                            return
-                        }
-                        self.addProximityEvent(deviceId: id, proximityEvent: pe, clProximity: clProximity)
-                    }
-                } else {
-                    // Do something when it doesn't need to be refreshed
-                }
-            }
+    // Trigger the proximity event
+    private func triggers(key: String, deviceId: String, distance: Double) {
+        let proximityEvent = ProximityEvent.init(deviceId: deviceId, distance: distance)
+        sendProximityEvent(deviceId: deviceId, proximityEvent: proximityEvent) {(_ proximityEvent) in
+            guard let pe = proximityEvent else {return}
+            self.beaconsTriggered[key]?[deviceId] = pe
         }
     }
     
-    // Remove the previous sent Proximity Event that was higher than a fixed time (Default: 5 minutes).
-    @objc
-    private func cleanTriggers() {
-        func clean(trigger: [String: ProximityEvent]) -> [String: ProximityEvent] {
```

Comment:
I got rid of this function, I see no use yet.
It was used to clear the trigger dictionary in case we won't detect Beacon BLE advertisement, based on a timer. But since timer are energy cost expensive I got rid of the timers, and this function was on stand-by.